### ISSUE TITLE
Feature: testing utilities

### ISSUE THREAD

**Issue by tkamenoko** [hattipjs/hattip#36]
Hattip has a lot of useful tools, but it lacks unit testing support. It sounds good to provide testing utilities like this:

```ts
const { fetch, cookies } = createTestClient(hattipHandler);
cookies.set("sessionId", "foobar");
const resp = await fetch("/api/example");
expect(resp.status).toEqual(200)
```

Prior arts:
- https://github.com/Xunnamius/next-test-api-route-handler : API testing tool for Next.js.
- https://github.com/encode/starlette : Python web framework. It has [`testClient`](https://www.starlette.io/testclient/) API for unit testing.

**Comment by cyco130** [comment]
Ooh, great idea.

It'd be useful for testing hattip itself too (and other tools based on it, like Rakkas).

I'll check the prior art, thanks for the suggestion.

**Comment by cyco130** [comment]
adapter-test released in 0.0.29
